js/ui/scheduler/shaders/ui.scheduler.current_time_shader.horizontal.js

```diff
@@ -1,35 +1,53 @@
-const Shader = require('./ui.scheduler.current_time_shader');
 import { getBoundingRect } from '../../../core/utils/position';
+import CurrentTimeShader from '../shaders/ui.scheduler.current_time_shader';
 
-const HorizontalCurrentTimeShader = Shader.inherit({
-    _renderShader: function() {
-        const groupCount = this._workspace.option('groupOrientation') === 'horizontal' ? this._workspace._getGroupCount() : 1;
+class HorizontalCurrentTimeShader extends CurrentTimeShader {
+    renderShader() {
+        const groupCount = this._workSpace.option('groupOrientation') === 'horizontal' ? this._workSpace._getGroupCount() : 1;
 
-        this._customizeShader(this._$shader, 0);
-
-        if(groupCount > 1) {
-            for(let i = 1; i < groupCount; i++) {
-                const $shader = this._createShader();
-                this._customizeShader($shader, 1);
-                this._shader.push($shader);
-            }
+        for(let i = 0; i < groupCount; i++) {
+            const $shader = i === 0 ? this._$shader : this.createShader();
+            this._workSpace.isGroupedByDate() ? this._customizeGroupedByDateShader($shader, i) : this._customizeShader($shader, i);
+            i !== 0 && this._shader.push($shader);
         }
-    },
+    }
+
+    _customizeShader($shader, groupIndex) {
+        const shaderWidth = this._workSpace.getIndicationWidth();
+
+        this._applyShaderWidth($shader, shaderWidth);
+
+        $shader.css('left', this._workSpace._getCellCount() * this._workSpace.getCellWidth() * (groupIndex >= 1 ? 1 : 0));
+    }
 
-    _customizeShader: function($shader, groupIndex) {
-        let shaderWidth = this._workspace.getIndicationWidth();
+    _applyShaderWidth($shader, width) {
         const maxWidth = getBoundingRect(this._$container.get(0)).width;
 
-        if(shaderWidth > maxWidth) {
-            shaderWidth = maxWidth;
+        if(width > maxWidth) {
+            width = maxWidth;
         }
 
-        if(shaderWidth > 0) {
-            $shader.width(shaderWidth);
+        if(width > 0) {
+            $shader.width(width);
         }
+    }
+
+    _customizeGroupedByDateShader($shader, groupIndex) {
+        const cellCount = this._workSpace.getIndicationCellCount();
+        const integerPart = Math.floor(cellCount);
+        const fractionPart = cellCount - integerPart;
+        const isFirstShaderPart = groupIndex === 0;
+
+        const shaderWidth = isFirstShaderPart ? this._workSpace.getIndicationWidth() : fractionPart * this._workSpace.getCellWidth();
+
+        this._applyShaderWidth($shader, shaderWidth);
+        this.applyShaderMargin($shader);
+
+        const shaderLeft = isFirstShaderPart ? this._workSpace._getCellCount() * this._workSpace.getCellWidth() * groupIndex :
+            this._workSpace.getCellWidth() * integerPart * this._workSpace._getGroupCount() + groupIndex * this._workSpace.getCellWidth();
```

Comment:
Too complex operation for the trenary operator.
`this._workSpace` should be cached to the local var.